numbat-cli/src/main.rs

```diff
@@ -351,123 +350,71 @@ impl Cli {
         rl: &mut Editor<NumbatHelper, DefaultHistory>,
         interactive: bool,
     ) -> Result<()> {
-        let mut session_history = SessionHistory::default();
+        let mut cmd_runner =
+            CommandRunner::<Editor<NumbatHelper, DefaultHistory>>::new_all_disabled()
+                .enable_print_markup(|m| println!("{}", ansi_format(m, true)))
+                .enable_clear(|rl| match rl.clear_screen() {
+                    Ok(_) => CommandControlFlow::Continue,
+                    Err(_) => CommandControlFlow::Return,
+                })
+                .enable_save(SessionHistory::default())
```

Comment:
What is the reasoning for a `session_history` argument here? Could this be `.enable_save()`? Or do we want frontends to be able to customize the session history object before passing it in for some reason?

Reply:
Yes, I wanted to leave the door open to letting frontends customize the SessionHistory, although currently there is no customization point.